### PR TITLE
Update docs/README link and folder name

### PR DESCRIPTION
**Pull request by alexandraj777**
I noticed a few small typos in the README and the docs and wanted to help to fix them. I'm not sure how contributing changes to projects like this normally works, so please let me know if I should change anything to do with my branching, forking, etc. Thanks!

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
 
-To build the documents locally, we need to first install [docker](docker.com).
+To build the documents locally, we need to first install [docker](https://docker.com).
 Then use the following commands to clone and
 build the documents.
 
 ```bash
 git clone --recursive https://github.com/apache/incubator-mxnet.git
-cd mxnet && make docs
+cd incubator-mxnet && make docs
```
Comment (piiswrong) [inline docs/README.md]: This seems to be happening at a lot of places after we moved repo. Should we change to 
`git clone --recursive https://github.com/apache/incubator-mxnet.git mxnet` everywhere? @kevinthesun

Comment (alexandraj777) [reply docs/README.md]: @piiswrong excellent suggestion! 

Let me know if you've like me to remove this particular change so you can defer it to a larger discussion/pr.

Comment (piiswrong) [reply docs/README.md]: This is all the places where this is needed.
```
c4b301d74d83:mxnet junyuanx$ git grep "https://github.com/apache/incubator-mxnet.git"
docs/README.md:git clone --recursive https://github.com/apache/incubator-mxnet.git
docs/build_version_doc/AddVersion.py:                                        'git clone --recursive https://github.com/apache/incubator-mxnet.git')
docs/build_version_doc/AddVersion.py:                                        'git clone --recursive https://github.com/apache/incubator-mxnet.git '
docs/build_version_doc/build_all_version.sh:mxnet_url="https://github.com/apache/incubator-mxnet.git"
```
Would you like to do the change at these 3 places?

Comment (alexandraj777) [reply docs/README.md]: Yeah I'd be happy to

File: docs/build_version_doc/build_all_version.sh [hunk 2]
```diff
@@ -24,7 +24,7 @@
 tag_list="0.11.0.rc3 master"
 
 mxnet_url="https://github.com/apache/incubator-mxnet.git"
-mxnet_folder="apache_mxnet"
+mxnet_folder="mxnet"
```
Comment (alexandraj777) [inline docs/build_version_doc/build_all_version.sh]: cc @piiswrong, I didn't know if this folder was purposely different from `mxnet` to avoid conflicts

Comment (piiswrong) [reply docs/build_version_doc/build_all_version.sh]: This is not part of doc

Comment (alexandraj777) [reply docs/build_version_doc/build_all_version.sh]: Oh duh >.<

**Comment by alexandraj777** [comment]
Thanks for reviewing @piiswrong!

**Pull request merged**